queries/roc/highlights.scm

```diff
@@ -0,0 +1,194 @@
+;---Most generic types---
+(module) @module
+
+(identifier) @variable
+
+(concrete_type) @type
+
+;---annotations----
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @type))
+
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @function)
+  (function_type))
+
+;----decleration types----
+(value_declaration
+  (decl_left
+    (identifier_pattern
+      (identifier) @variable.parameter)))
+
+;---records----
+(field_name) @variable.member
+
+(record_field_pattern
+  (_
+    (identifier) @variable))
+
+;matches the second identifier and all subsequent ones
+(field_access_expr
+  (identifier) @variable.member)
+
+;----comments----
+[
+  (line_comment)
+  (doc_comment)
+] @comment
+
+;-----Punctuation----
+[
+  "?"
+  (arrow)
+  (back_arrow)
+  (backslash)
+] @punctuation.delimiter
+
+[
+  ","
+  ":"
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "{"
+  "}"
+  "["
+  "]"
+] @punctuation.bracket
+
+[
+  "|"
+  "&"
+  (operator)
+  (wildcard_pattern)
+] @operator
+
+[
+  "if"
+  "then"
+  "else"
+] @keyword.conditional
+
+[
+  (implements)
+  (when)
+  (is)
+  "as"
+  (to)
+] @keyword
+
+;----headers-----
+(interface_header
+  (name) @type.definition)
+
+(imports
+  (imports_entry
+    (module) @module))
+
+(packages
+  (record_pattern
+    (record_field_pattern
+      (field_name) @module)))
+
+(app_name) @string
+
+(import_path) @string
+
+[
+  "app"
+  "packages"
+  "imports"
+  "provides"
+  "interface"
+  "exposes"
+  "expect"
+  (import_as)
+] @keyword
+
+(backpassing_expr
+  assignee: (identifier_pattern
+    (identifier) @variable.parameter))
+
+(value_declaration
+  (decl_left
+    (identifier_pattern
+      (identifier) @function))
+  (expr_body
+    (anon_fun_expr)))
+
+;----tags----
+(tags_type
+  (apply_type
+    (concrete_type) @constructor))
+
+(tag) @constructor
+
+(opaque_tag) @constructor
+
+;-----builtins----
+(variable_expr
+  (module) @module
+  (identifier) @constant.builtin
+  (#eq? @constant.builtin "true")
+  (#eq? @module "Bool"))
+
+(variable_expr
+  (module) @module
+  (identifier) @constant.builtin
+  (#eq? @constant.builtin "false")
+  (#eq? @module "Bool"))
+
+"dbg" @constant.builtin
+
+;----function invocations ----
+(function_call_expr
+  caller: (variable_expr
+    (identifier) @function.call))
+
+(function_call_expr
+  caller: (field_access_expr
+    (identifier) @function.call .))
+
+(bin_op_expr
+  (operator
+    "|>") @operator
+  (variable_expr
+    (identifier) @function))
+
+;----function arguments----
+(argument_patterns
+  (identifier_pattern
+    (identifier) @variable.parameter))
+
+(argument_patterns
+  (_
+    (identifier_pattern
+      (identifier) @variable.parameter)))
+
+(argument_patterns
+  (_
+    (_
+      (identifier_pattern
+        (identifier) @variable.parameter))))
+
+;-----consts-----
+[
+  (int)
+  (uint)
+  (iint)
+  (xint)
+  (natural)
+] @constant
+
+[
+  (decimal)
+  (float)
+] @constant
+
+(string) @string
+
+(char) @constant
```

Comment:
Maybe `@character` (or `@character.special`, don't know this language well) for this